All files access switch greyed out when trying to grant files permissions

### Description of the issue
all files access switch greyed out when trying to grant files permissions

### Reproducer
open a fresh install of syncthing fork and try to grant files permissions

### Version Information
    App Version: 1.9.0.2
    Syncthing Version: v1.9.0
    Android Version: Android 11 full release
    Device manufacturer: Google
    Device model: Pixel 2 G011A

<details>

```
### Device platform info
[ro.product.board]: [walleye]
[ro.product.brand]: [google]
[ro.product.build.date]: [Wed Jul 29 23:26:21 UTC 2020]
[ro.product.build.date.utc]: [1596065181]
[ro.product.build.fingerprint]: [google/walleye/walleye:11/RP1A.200720.009/6720564:user/release-keys]
[ro.product.build.id]: [RP1A.200720.009]
[ro.product.build.tags]: [release-keys]
[ro.product.build.type]: [user]
[ro.product.build.version.incremental]: [6720564]
[ro.product.build.version.release]: [11]
[ro.product.build.version.release_or_codename]: [11]
[ro.product.build.version.sdk]: [30]
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: [arm64-v8a]
[ro.product.device]: [walleye]
[ro.product.first_api_level]: [26]
[ro.product.locale]: [en-US]
[ro.product.manufacturer]: [Google]
[ro.product.model]: [Pixel 2]
[ro.product.name]: [walleye]
[ro.product.odm.brand]: [google]
[ro.product.odm.device]: [walleye]
[ro.product.odm.manufacturer]: [Google]
[ro.product.odm.model]: [Pixel 2]
[ro.product.odm.name]: [walleye]
[ro.product.product.brand]: [google]
[ro.product.product.device]: [walleye]
[ro.product.product.manufacturer]: [Google]
[ro.product.product.model]: [Pixel 2]
[ro.product.product.name]: [walleye]
[ro.product.system.brand]: [google]
[ro.product.system.device]: [walleye]
[ro.product.system.manufacturer]: [Google]
[ro.product.system.model]: [Pixel 2]
[ro.product.system.name]: [walleye]
[ro.product.system_ext.brand]: [google]
[ro.product.system_ext.device]: [walleye]
[ro.product.system_ext.manufacturer]: [Google]
[ro.product.system_ext.model]: [Pixel 2]
[ro.product.system_ext.name]: [walleye]
[ro.product.vendor.brand]: [google]
[ro.product.vendor.device]: [walleye]
[ro.product.vendor.manufacturer]: [Google]
[ro.product.vendor.model]: [Pixel 2]
[ro.product.vendor.name]: [walleye]
```

### Android Log
[android log.txt](https://github.com/Catfriend1/syncthing-android/files/5190810/android.log.txt)

![image](https://user-images.githubusercontent.com/70977962/92521708-979b6500-f215-11ea-8456-14eb2e90de4e.png)

</details>

Oh I don't think that's needed. The app works normally _if_ you open it from the notifications, and it does sync, though I believe, the folders have become read only so it's a one way sync.

So the export menu can be accessed, but when you press the export button, that's when it fails because I think the directory it's writing the export to is also read only.

I added a few logs and screenshots below. I hope that helps.

It starts with this repeated many times
```
09-09 07:28:23.769 10068 30638 D Util    : runShellCommandGetOutput: ps
09-09 07:28:23.786 10068 30638 I Util    : runShellCommandGetOutput: Exited with code 0
```
Then
```
09-09 07:28:23.819 10068 26094 I SyncthingNativeCode: [G6XDM] WARNING: Database failed to stop within 10s
09-09 07:28:23.819 10068 26094 I SyncthingNativeCode: [G6XDM] INFO: Exiting
09-09 07:28:23.838 10068 30638 D Util    : runShellCommandGetOutput: ps
09-09 07:28:23.854 10068 30638 I Util    : runShellCommandGetOutput: Exited with code 0
09-09 07:28:23.854 10068 30638 D SyncthingRunnable: killSyncthing: Complete.
09-09 07:28:24.014 10068 26090 I SyncthingRunnable: Syncthing was shut down normally via API or SIGKILL. Exit code = 0
09-09 07:28:24.014 10068 30638 D SyncthingService: Finished mSyncthingRunnableThread.
09-09 07:28:24.020  3676 28756 E MediaProvider: Creating a non-default top level directory or deleting an existing one is not allowed!
09-09 07:28:24.020  3676 28753 E MediaProvider: Creating a non-default top level directory or deleting an existing one is not allowed!
09-09 07:28:24.025 10068 30638 W SyncthingService: Failed to export config
09-09 07:28:24.025 10068 30638 W SyncthingService: java.io.FileNotFoundException: /storage/emulated/0/backups/syncthing/config.xml: open failed: ENOENT (No such file or directory)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at libcore.io.IoBridge.open(IoBridge.java:492)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at java.io.FileOutputStream.<init>(FileOutputStream.java:236)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at com.google.common.io.Files$FileByteSink.openStream(Files.java:193)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at com.google.common.io.Files$FileByteSink.openStream(Files.java:181)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at com.google.common.io.ByteSource.copyTo(ByteSource.java:265)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at com.google.common.io.Files.copy(Files.java:326)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at com.nutomic.syncthingandroid.service.SyncthingService.exportConfig(SyncthingService.java:859)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at com.nutomic.syncthingandroid.activities.SettingsActivity$SettingsFragment$ExportConfigTask.doInBackground(SettingsActivity.java:1019)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at com.nutomic.syncthingandroid.activities.SettingsActivity$SettingsFragment$ExportConfigTask.doInBackground(SettingsActivity.java:1002)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at android.os.AsyncTask$3.call(AsyncTask.java:394)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at java.lang.Thread.run(Thread.java:923)
09-09 07:28:24.025 10068 30638 W SyncthingService: Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at libcore.io.Linux.open(Native Method)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at libcore.io.ForwardingOs.open(ForwardingOs.java:166)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at libcore.io.BlockGuardOs.open(BlockGuardOs.java:254)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at libcore.io.ForwardingOs.open(ForwardingOs.java:166)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7542)
09-09 07:28:24.025 10068 30638 W SyncthingService:  at libcore.io.IoBridge.open(IoBridge.java:478)
09-09 07:28:24.025 10068 30638 W SyncthingService:  ... 13 more
09-09 07:28:24.026 10068 30638 E SyncthingService: exportConfig: Failed to export SharedPreferences #1
09-09 07:28:24.026 10068 30638 E SyncthingService: java.io.FileNotFoundException: /storage/emulated/0/backups/syncthing/sharedpreferences.dat: open failed: ENOENT (No such file or directory)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at libcore.io.IoBridge.open(IoBridge.java:492)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at java.io.FileOutputStream.<init>(FileOutputStream.java:236)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at java.io.FileOutputStream.<init>(FileOutputStream.java:186)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at com.nutomic.syncthingandroid.service.SyncthingService.exportConfig(SyncthingService.java:876)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at com.nutomic.syncthingandroid.activities.SettingsActivity$SettingsFragment$ExportConfigTask.doInBackground(SettingsActivity.java:1019)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at com.nutomic.syncthingandroid.activities.SettingsActivity$SettingsFragment$ExportConfigTask.doInBackground(SettingsActivity.java:1002)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at android.os.AsyncTask$3.call(AsyncTask.java:394)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at java.lang.Thread.run(Thread.java:923)
09-09 07:28:24.026 10068 30638 E SyncthingService: Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at libcore.io.Linux.open(Native Method)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at libcore.io.ForwardingOs.open(ForwardingOs.java:166)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at libcore.io.BlockGuardOs.open(BlockGuardOs.java:254)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at libcore.io.ForwardingOs.open(ForwardingOs.java:166)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7542)
09-09 07:28:24.026 10068 30638 E SyncthingService:  at libcore.io.IoBridge.open(IoBridge.java:478)
09-09 07:28:24.026 10068 30638 E SyncthingService:  ... 10 more
09-09 07:28:24.026 10068 30638 D SyncthingService: exportConfig: Exporting index database
```

then a bunch of 
```
09-09 07:28:24.028 10068 30638 E SyncthingService: Failed to copy file '/data/user/0/com.github.catfriend1.syncthingandroid/files/index-v0.14.0.db' to '/storage/emulated/0/backups/syncthing/index-v0.14.0.db'
```
for several ldb files

and finally,
```
09-09 07:28:24.094 10068 30638 D SyncthingService: exportConfig END
```

I then manually created the `backups/syncthing` directory and retried the export, and the logs were mostly the same, except for a few new error logs like this
```
09-09 07:37:41.976  3676 29505 W MediaProvider: Forgot to handle a top level directory in getContentUriForFile?
09-09 07:37:41.980  3676 29505 E MediaProvider: insertFileIfNecessary failed
09-09 07:37:41.980  3676 29505 E MediaProvider: java.lang.IllegalArgumentException: Primary directory backups not allowed for content://media/external_primary/file; allowed directories are [Download, Documents]
09-09 07:37:41.980  3676 29505 E MediaProvider:   at com.android.providers.media.MediaProvider.ensureFileColumns(MediaProvider.java:2672)
09-09 07:37:41.980  3676 29505 E MediaProvider:   at com.android.providers.media.MediaProvider.ensureUniqueFileColumns(MediaProvider.java:2337)
09-09 07:37:41.980  3676 29505 E MediaProvider:   at com.android.providers.media.MediaProvider.insertFile(MediaProvider.java:2947)
09-09 07:37:41.980  3676 29505 E MediaProvider:   at com.android.providers.media.MediaProvider.insertInternal(MediaProvider.java:3470)
09-09 07:37:41.980  3676 29505 E MediaProvider:   at com.android.providers.media.MediaProvider.insert(MediaProvider.java:3198)
09-09 07:37:41.980  3676 29505 E MediaProvider:   at com.android.providers.media.MediaProvider.insertFileForFuse(MediaProvider.java:6625)
09-09 07:37:41.980  3676 29505 E MediaProvider:   at com.android.providers.media.MediaProvider.insertFileIfNecessaryForFuse(MediaProvider.java:6708)
09-09 07:37:41.981 10068 30638 E SyncthingService: Failed to copy file '/data/user/0/com.github.catfriend1.syncthingandroid/files/index-v0.14.0.db/013929.ldb' to '/storage/emulated/0/backups/syncthing/index-v0.14.0.db'
```

![Screenshot_20200909-072448](https://user-images.githubusercontent.com/536147/92594893-82ecba80-f271-11ea-9056-17b8afe0c784.png)
![Screenshot_20200909-072825](https://user-images.githubusercontent.com/536147/92594889-81bb8d80-f271-11ea-94d0-f8a897ec50ee.png)

@auanasgheps 

I've made a new gplay build including the ALL FILES ACCESS permission but Google still refuses the upload to the play store.

Build log:
```
Uploading App Bundle: 21% complete
Uploading App Bundle: 41% complete
Uploading App Bundle: 62% complete
Uploading App Bundle: 82% complete

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:publishGplayBundle'.
> A failure occurred while executing com.github.triplet.gradle.play.tasks.PublishBundle$BundleUploader
   > 400 Bad Request
     PUT https://www.googleapis.com/upload/androidpublisher/v3/applications/com.github.catfriend1.syncthingandroid/edits/XXXX/bundles?uploadType=resumable&upload_id=YYYY
     {
       "code" : 400,
       "errors" : [ {
         "domain" : "global",
         "message" : "Cannot use MANAGE_EXTERNAL_STORAGE permission",
         "reason" : "badRequest"
       } ],
       "message" : "Cannot use MANAGE_EXTERNAL_STORAGE permission",
       "status" : "INVALID_ARGUMENT"
     }
```

--- a/app/src/gplay/AndroidManifest.xml
+++ b/app/src/gplay/AndroidManifest.xml
@@ -6,6 +6,5 @@
     <uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" tools:node="remove"/>
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" tools:node="remove"/>
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" tools:node="remove"/>
-    <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE" tools:node="remove"/>
 
 </manifest>

--- a/app/versions.gradle
+++ b/app/versions.gradle
@@ -2,5 +2,5 @@ ext {
     versionMajor = 1
     versionMinor = 16
     versionPatch = 0
-    versionWrapper = 1
+    versionWrapper = 3
 }

